src/utils/command.rs

```diff
@@ -41,75 +41,189 @@
+///    - `default` - the same as the unspecified parser. It only puts all text
+///    after the first space into the first argument, which must implement
+///    [`FromStr`].
+///
+/// ### Example
+/// ```
+/// use teloxide::utils::command::BotCommand;
+///
+/// #[derive(BotCommand, PartialEq, Debug)]
+/// #[command(rename = "lowercase")]
+/// enum Command {
+///     Text(String),
+/// }
+///
+/// let command =
+///     AdminCommand::parse("/text hello my dear friend!", "").unwrap();
+/// assert_eq!(command, Command::Text("hello my dear friend!".to_string()));
+/// ```
+///
+///  - `split` - separates a messsage by a given separator (the default is the
+///    space character) and parses each part into the corresponding arguments,
+///    which must implement [`FromStr`].
+///
+/// ### Example
+/// ```
+/// use teloxide::utils::command::BotCommand;
+///
+/// #[derive(BotCommand, PartialEq, Debug)]
+/// #[command(rename = "lowercase", parser = "split")]
+/// enum Command {
+///     Nums(u8, u16, i32),
+/// }
+///
+/// let command = AdminCommand::parse("/nums 1 32 -5", "").unwrap();
+/// assert_eq!(command, Command::Nums(1, 32, -5));
+/// ```
+///
+/// 5. `#[command(separator = "sep")]`
+/// Specify separator used by the `split` parser. It will be ignored when
+/// accompanied by another type of parsers.
+///
+/// ### Example
+/// ```
+/// use teloxide::utils::command::BotCommand;
+///
+/// #[derive(BotCommand, PartialEq, Debug)]
+/// #[command(rename = "lowercase", parser = "split", separator = "|")]
+/// enum Command {
+///     Nums(u8, u16, i32),
+/// }
+///
+/// let command = AdminCommand::parse("/nums 1|32|5", "").unwrap();
+/// assert_eq!(command, Command::Nums(1, 32, 5));
+/// ```
+///
 /// ## Variant attributes
+/// All variant attributes override the corresponding `enum` attributes.
+///
 ///  1. `#[command(rename = "rule")]`
 /// Rename one command by a rule. Allowed rules are `lowercase`, `%some_name%`,
 /// where `%some_name%` is any string, a new name.
 ///
-///  2. `#[command(prefix = "prefix")]`
-/// Change a prefix for one command (the default is `/`).
+///  2. `#[command(parser = "parser")]`
+/// One more option is available for variants.
+///    - `custom_parser` - your own parser of the signature `fn(String) ->
+///    Result<Tuple, ParseError>`, where `Tuple` corresponds to the variant's
+/// arguments.
 ///
-///  3. `#[command(description = "description")]`
-/// Add a description of one command.
+/// ### Example
+/// ```
+/// use teloxide::utils::command::{BotCommand, ParseError};
+///
+/// fn accept_two_digits(input: String) -> Result<(u8), ParseError> {
+///     match input.len() {
+///         2 => {
+///             let num =
+///                 input.parse().map_err(|_| ParseError::IncorrectFormat)?;
+///             Ok((num))
+///         }
+///         len => Err(ParseError::Custom(format!(
+///             "Only 2 digits allowed, not {}",
+///             len
+///         ))),
+///     }
+/// }
```

Comment:
Where is this function used?

Reply:
Oh, it must be `#[command(parser="accept_two_digits")]` at enum variant